### PR TITLE
Add core types enums description to extension api json

### PR DESCRIPTION
**Pull request by piiertho**
This adds description of core types enums to extension api json.  
Enums are described the same way as global and common classes enums.

### CODE REVIEW

File: core/variant/variant.h [hunk 1]
```diff
@@ -715,6 +715,10 @@ class Variant {
 	static bool has_constant(Variant::Type p_type, const StringName &p_value);
 	static Variant get_constant_value(Variant::Type p_type, const StringName &p_value, bool *r_valid = nullptr);
 
+	static void get_enums_for_type(Variant::Type p_type, List<StringName> *p_enums);
```
Comment (reduz) [inline core/variant/variant.h:718]: So I have a doubt here, is this not redundant with constants? Would it not be better to simply have a method like `get_enum_constants` besides `get_enums_for_type` ?

Comment (piiertho) [reply core/variant/variant.h:718]: I don't think so. There is a hierarchy level in more with enum.  
It is kind of namespaced constants, but I still need to specify for which "namespace" I want to list.

**Review by reduz — APPROVED** [review]

**Comment by akien-mga** [comment]
Probably needs a rebase as we changed some core code like Map -> HashMap (or RBMap depending on the need).

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**